app/custom_validator.py

```diff
@@ -0,0 +1,39 @@
+from connexion.decorators.response import ResponseValidator
+from connexion.json_schema import Draft4ResponseValidator
+from jsonschema import Draft7Validator
+from jsonschema.validators import extend
+
+
+def get_draft4_custom_validators():
+    """
+    Copies over the custom validators from connexion's Draft4ResponseValidator and returns them in a new dictionary
+    """
+    custom_validator_names = ["type", "enum", "required", "writeOnly", "x-writeOnly"]
+    custom_validators = {}
+
+    for name in custom_validator_names:
+        custom_validators[name] = Draft4ResponseValidator.VALIDATORS.get(name)
+
+    return custom_validators
+
+
+def get_custom_validator():
+    custom_validators = get_draft4_custom_validators()
+    CustomDraft7Validator = extend(
+        Draft7Validator, {**custom_validators, "x-propertyNames": Draft7Validator.VALIDATORS.get("propertyNames")}
+    )
+    return CustomDraft7Validator
```

Comment:
All this is unnecessarily complex and not entirely correct. Earlier, we already agreed that the Schema is not Draft 7, but rather Draft 4 (standard for OpenAPI) with our custom extensions.

All we need to add to _Draft4ResponseValidator_ is the _propertyNames_ validator from Draft 7 hooked to _x-propertyNames_. That can be simply achieved by extending the original one as the previous solution did.

```suggestion
CustomDraft4ResponseValidator = extend(
    Draft4ResponseValidator, {"x-propertyNames": Draft7Validator.VALIDATORS.get("propertyNames")}
)
```